lib/action_controller/serialization.rb

```diff
@@ -43,6 +44,12 @@ def get_serializer(resource, options = {})
       end
     end
 
+    def _handle_errors(serializable_resource)
+      return unless serializable_resource.errors?
+      return unless (200..299).cover?(response.status)
+      response.status = 422
```

Comment:
I like the implementation of this (specifically the case/when logic). as long as within each when block, there was a method call to `handle_whatever_error`, it would allow overriding in the controller, which I think would be a nice comprise to allow flexability